providers/terraform-provider-csbdynamodbns/csbdynamodbns/resource_dynamodbns_instance.go

```diff
@@ -0,0 +1,74 @@
+package csbdynamodbns
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/smithy-go/ptr"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	awsAccessKeyIDKey     = "access_key_id"
+	awsSecretAccessKeyKey = "secret_access_key"
+)
+
+func resourceDynamoDBNSInstance() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			awsAccessKeyIDKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			awsSecretAccessKeyKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+		CreateContext: setResourceID,
+		UpdateContext: setResourceID,
+		ReadContext:   setResourceID,
+		DeleteContext: resourceDynamoDBMaintenanceDelete,
+		Description:   "Handles DynamoDB namespace housekeeping",
+	}
+}
+
+func setResourceID(_ context.Context, d *schema.ResourceData, _ any) diag.Diagnostics {
+	keyID := d.Get(awsAccessKeyIDKey).(string)
+	d.SetId(keyID)
+	return nil
+}
+
+func resourceDynamoDBMaintenanceDelete(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	settings := m.(*dynamoDBNamespaceSettings)
+	client, err := settings.GetClient(ctx, d.Get(awsAccessKeyIDKey).(string), d.Get(awsSecretAccessKeyKey).(string))
```

Comment:
Hypothetically, the credentials may be regenerated, so they are passed as part of ResourceData, which is only available at the time of this call.

Reply:
Indeed, but in the context of the broker, where this code is intended to be used, the credentials won't be changed as the terraform workspace is a created each time we perform an operation. It's not a big deal, but it would be nice to have consistency across all our providers.